internal/consensus/msgs.go

```diff
@@ -220,9 +220,11 @@ type VoteMessage struct {
 
 func (*VoteMessage) TypeTag() string { return "tendermint/Vote" }
 
-// ValidateBasic performs basic validation.
+// ValidateBasic checks whether the vote within the message is well-formed.
 func (m *VoteMessage) ValidateBasic() error {
-	return m.Vote.ValidateBasic()
+	// Here we validate votes with vote extensions, since we require vote
+	// extensions to be sent in vote messages during consensus.
```

Comment:
Is it also the case with prevotes?
I mean... may they also come in a `VoteMessage`